src/assert.js

```diff
@@ -15,25 +15,38 @@ QUnit.assert = Assert.prototype = {
 		}
 	},
 
-	// Increment this Test's semaphore counter, then return a single-use function that
+	// Increment this Test's semaphore counter, then return a function that
 	// decrements that counter a maximum of once.
-	async: function() {
+	async: function( count ) {
 		var test = this.test,
-			popped = false;
+			popped = false,
+			acceptCallCount = count;
+
+		if ( typeof acceptCallCount === "undefined" ) {
+			acceptCallCount = 1;
+		}
 
 		test.semaphore += 1;
 		test.usedAsync = true;
 		pauseProcessing();
 
 		return function done() {
-			if ( !popped ) {
-				test.semaphore -= 1;
-				popped = true;
-				resumeProcessing();
-			} else {
-				test.pushFailure( "Called the callback returned from `assert.async` more than once",
+			acceptCallCount -= 1;
+
+			if ( acceptCallCount > 0 ) {
+				pauseProcessing();
```

Comment:
> If `pauseProcessing()` do not called that delay aggregate time is `testTimeout` time when all `done()` finished. The `testTimeout` is too short for all `done` called. So When `done()` called, reset `setTimeout`.

`config.testTimeout` always counts from the start of each test; see http://api.qunitjs.com/QUnit.config/ ("a global timeout in milliseconds after which all tests will fail… to prevent the testrunner getting stuck").